src/flow.ts

```diff
@@ -237,6 +244,11 @@ export class Flow {
   localFlags: LocalFlags[] = [];
   /** Field flags on `this`. Constructors only. */
   thisFieldFlags: Map<Field,FieldFlags> | null = null;
+  /** type narrow */
+  narrowedTypes: NarrowedTypeMap = new NarrowedTypeMap();
+  /** conditional type narrow, eg if (condi) */
+  conditionalNarrowedType: TypeNarrowChecker = new TypeNarrowChecker();
```

Comment:
it will be great init checker and narrowedTypes lazily. Not every program uses `instanceof`. And it also add overhead for cloning in `fork` method